components/AppHeader.js

```diff
@@ -0,0 +1,188 @@
+import { View, Text, StyleSheet, SafeAreaView, Animated } from "react-native";
+import React, { useEffect, useRef, useState } from "react";
+import { MaterialIcons } from "@expo/vector-icons";
+import { FontAwesome } from "@expo/vector-icons";
+import SearchModal from "./SearchModal";
+import { useStore } from "../StoreProvider";
+
+const AppHeader = ({ weatherDetails }) => {
+  const { getUserLocation, setShowModal, isAnimating } = useStore();
+  const [date, setDate] = useState("");
+
+  function handleFormateDate() {
```

Comment:
<img width="844" alt="Screenshot 2024-12-26 at 8 34 06 AM" src="https://github.com/user-attachments/assets/97ce7cc8-68a9-4572-baea-d7d9611bc54e" />
gpt suggested this, have a look. 

lot of times date format locally is different. 
in north america they put month before date. 

the intl. local does the formatting for you according to browser location. ( not geo location, but default configs)
in romania, date is shown different, etc. 

this type of handling gets extra marks.